Paper: Echodataflow: Recipe-based Fisheries Acoustics Workflow Orchestration

valentina-s opened the pull request:
If you are creating this PR in order to submit a draft of your paper, please name your PR with `Paper: <title>`. An editor will then add a `paper` label and GitHub Actions will be run to check and build your paper.

See the [project readme](https://github.com/scipy-conference/scipy_proceedings#preview-your-paper) for more information.

_Editor_: <!--editor-->Meghann Agarwal @mepa<!--end-editor-->

_Reviewers_: <!--reviewers-list-->
- ~~Ruman Dagol @gol-danRuman~~
- Nadia Tahiri @TahiriNadia 
- Meghann Agarwal @mepa
<!--end-reviewers-list-->

valentina-s committed:
initial commit

valentina-s committed:
fixing project id

valentina-s committed:
updating title in main

rowanc1 commented on papers/valentina_staneva/myst.yml:
```diff
@@ -0,0 +1,52 @@
+  # Update this to match `scipy-2024-<folder>` the folder should be `<firstname_surname>`
+  id: scipy-2024-valentina_staneva
+  # Ensure your title is the same as in your `main.md`
+  title: echodataflow: Configurable, Reproducible, and Scalable Fisheries Acoustics Workflows with Prefect
```
For the YAML, this needs quotes:

```suggestion
  title: "echodataflow: Configurable, Reproducible, and Scalable Fisheries Acoustics Workflows with Prefect"
```

valentina-s committed:
adding quotes

rowanc1 commented on papers/valentina_staneva/myst.yml:
```diff
@@ -0,0 +1,52 @@
+      affiliations:
+        - eScience Institute, University of Washington
+    - name: Landung (Don) Setiawan
+      email:landungs@uw.edu
```
Looks like another problem in YAML (note the space):

```suggestion
      email: landungs@uw.edu
```

valentina-s committed:
trying out reference

valentina-s committed:
fixing white space

valentina-s committed:
adding figure properly

valentina-s committed:
adding back vertical line

valentina-s committed:
fixing abstract

valentina-s committed:
adding caption

valentina-s committed:
removing figure

valentina-s committed:
fig reference

valentina-s committed:
cleaning bib to trigger rebuild

valentina-s committed:
Adding figures

valentina-s committed:
fixing typos

valentina-s committed:
fixing task progress

valentina-s committed:
adding a dradt for deployment and client

valentina-s committed:
more details on deployment+cli

valentina-s committed:
adding figures at their places

valentina-s committed:
removing notebook_start

valentina-s committed:
adding introductory sections

valentina-s committed:
data collection

valentina-s committed:
fisheries use case

valentina-s committed:
adding task progress

valentina-s committed:
fixing title

valentina-s committed:
add references

valentina-s committed:
add references

ameyxd commented:
@valentina-s thanks for your submission! Can you update the DOIs for "echopy_2024" and "korneliussen_lsss_2006"? We can run the checks again after. 

you may be able to add citation keys you want to ignore if their DOIs don't exist in myst.yml under error_rules:

  ```
error_rules:
    - rule: doi-exists
      severity: ignore
      keys:
        - abc
        - def01
```

valentina-s committed:
adding some references

valentina-s committed:
other workflow tools and references

valentina-s committed:
language edits

valentina-s committed:
language edits

Commit:
update survey section

Commit:
update introduction

Commit:
workflow details

Commit:
workflow details

valentina-s committed:
adding figures

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

Commit:
fixing figure reference

valentina-s committed:
adding all figures

Commit:
update text to frequency differencing

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

valentina-s committed:
update data collection image

Commit:
major editing

valentina-s committed:
references

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

Commit:
NOAA

valentina-s committed:
fixing comma

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

valentina-s committed:
fixing braces

valentina-s committed:
fixing comma

valentina-s committed:
fixing repeated reference

valentina-s committed:
fixing missing references

valentina-s committed:
references

Commit:
fixing reference

Commit:
resolve @echodataflow

valentina-s committed:
fixing capital letter

Commit:
fix first figure

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

Commit:
last changed at Jun 8, 2024 2:20 AM, pushed by Valentina Staneva

Commit:
last changed at Jun 8, 2024 2:20 AM, pushed by Valentina Staneva

Commit:
last changed at Jun 8, 2024 2:20 AM, pushed by Valentina Staneva

Commit:
fixinf fig reference

Commit:
@ -> #

valentina-s committed:
ignoring missing DOI

valentina-s committed:
Merge branch 'echodataflow' of https://github.com/valentina-s/scipy_proceedings into echodataflow

valentina-s committed:
ignoring missing DOI

TahiriNadia commented:
Hi @valentina-s, 

I'm Nadia Tahiri. I will be one of the reviewers of your paper, and I will provide detailed comments and suggestions soon.

Cheers,

ameyxd commented:
> @valentina-s thanks for your submission! Can you update the DOIs for "echopy_2024" and "korneliussen_lsss_2006"? We can run the checks again after. 
> 
> 
> 
> you may be able to add citation keys you want to ignore if their DOIs don't exist in myst.yml under error_rules:
> 
> 
> 
>   ```
> 
> error_rules:
> 
>     - rule: doi-exists
> 
>       severity: ignore
> 
>       keys:
> 
>         - abc
> 
>         - def01
> 
> ```

Hi @valentina-s are you skipping any valid DOIs? We have fixed issues with failing checks for valid DOIs. please check and lmk. 🙂

mepa commented:
Hi @gol-danRuman and @TahiriNadia! In case a little extra time is needed, the initial complete review deadline has been extended to next [Wednesday, July 3rd](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#instructions-for-reviewers).

TahiriNadia commented:
Dear @valentina-s, I will list my comments one by one below in detail.

The submitted article explores the application of a recipe-based approach to dataflow orchestration within the fisheries context. This methodology aims to optimize data processing workflows in fisheries research and management by leveraging advanced scientific computing techniques. The article is highly relevant, well-structured, and written with clarity.

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+
+## Motivation
+Acoustic fisheries surveys and ocean observing systems collect terabytes of echosounder (water column sonar) data that require custom processing pipelines to obtain the distributions and abundance of fish and zooplankton in the ocean [@ncei_story_map]. The data are collected by sending an acoustic signal into the ocean which scatters from objects in the water column and the returning “echo” is re ...
```
Add a space before parentheses. e.g. Saildrone[@saildrone], DriX[@drix]

TahiriNadia replied:
What are the typical dimensions of acoustic fisheries survey data, and how do they vary across different data collection methods?

TahiriNadia replied:
What role does computational infrastructure play in adapting data workflows to different research needs and data volumes in acoustic fisheries surveys?

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+
+### Software Landscape
+Traditionally echosounder data processing pipelines are executed within a GUI-based software (e.g. Echoview[@harrison_echoviewR_2015], LSSS[@korneliussen_lsss_2006], ESP3[@ladroit_esp3_2020], Matecho[perrot_matecho_2018]). These software packages have been invaluable for onboard real-time visualization, as well as post-survey data screening and annotation. Some of them also support integration wi ...
```
Add a space before parentheses. Echoview[@harrison_echoviewR_2015], LSSS[@korneliussen_lsss_2006], ESP3[@ladroit_esp3_2020], Matecho[perrot_matecho_2018])

TahiriNadia replied:
Could you check the year of this citation software? echoviewR package [@echoviewr]

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+##### Data Storage
+
+Researchers are faced with decisions of where to store the data from experiments, intermediate products, and final results. Initially, data are usually stored on local hard drive storage associated with the instrument (which on some platforms may have limited capacity), but eventually, these data may be transferred to a data archive if one is maintained within the community. Some agencies (e.g. N ...
```
Could you provide the acronym for the National Centers for Environmental Information, and add a comma before it?

TahiriNadia replied:
transform (@wall_2016) to [@wall_2016]

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+### Software Landscape
+Traditionally echosounder data processing pipelines are executed within a GUI-based software (e.g. Echoview[@harrison_echoviewR_2015], LSSS[@korneliussen_lsss_2006], ESP3[@ladroit_esp3_2020], Matecho[perrot_matecho_2018]). These software packages have been invaluable for onboard real-time visualization, as well as post-survey data screening and annotation. Some of them also support integration wi ...
+ Over the last several years there has been substantial development of open source Python packages (PyEchoLab [@wall_pyecholab_2018], `echopype`[@lee_echopype_2021], `echopy`[@echopy_2024]), each providing common echosounder processing functionalities. Since echosounder instruments store the data in  binary, instrument-specific formats, the first stage requires parsing the raw data into a more co ...
```
Add a space before parentheses. `echopype`[@lee_echopype_2021], `echopy`[@echopy_2024] ... `numpy`[@numpy]

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+##### Computing Infrastructure
+
+With the growth of the echosounder datasets, researchers face challenges processing the data on their personal machines: both in terms of memory usage and computational time. A typical first attempt for resolution would be amend the workflow to process smaller chunks of the data and parallelize operations across multiple cores if available. However, today researchers are also presented with a mul ...
```
Remove the extra space before the comma in (e.g. , serverless options (e.g. Amazon Web Services Lamdba Functions, Google Cloud Functions, Microsoft Azure Functions). and double e.g. and missing ). Ensure to double-check this sentence for accuracy.

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+With the growth of the echosounder datasets, researchers face challenges processing the data on their personal machines: both in terms of memory usage and computational time. A typical first attempt for resolution would be amend the workflow to process smaller chunks of the data and parallelize operations across multiple cores if available. However, today researchers are also presented with a mul ...
+
+## Echodataflow Overview
+At the center of `echodataflow`'s design is the notion that a workflow can be configured through a set of recipes (`.yaml` files) that specify the pipeline, data storage, and logging structure. The idea draws inspiration from the Pangeo-Forge Project which facilitates the Extraction, Transformation, Loading (ETL) of earth science geospatial datasets from traditional repositories to analysis-ready ...
```
Could you check the year of this citation software? [@hdf5]

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+With the growth of the echosounder datasets, researchers face challenges processing the data on their personal machines: both in terms of memory usage and computational time. A typical first attempt for resolution would be amend the workflow to process smaller chunks of the data and parallelize operations across multiple cores if available. However, today researchers are also presented with a mul ...
+
+## Echodataflow Overview
+At the center of `echodataflow`'s design is the notion that a workflow can be configured through a set of recipes (`.yaml` files) that specify the pipeline, data storage, and logging structure. The idea draws inspiration from the Pangeo-Forge Project which facilitates the Extraction, Transformation, Loading (ETL) of earth science geospatial datasets from traditional repositories to analysis-ready ...
```
Add a comma after 'In [Figure %s ](#fig:echodataflow_overview)'

TahiriNadia replied:
Can you elaborate on how echodataflow compares and contrasts with Pangeo-Forge in terms of its approach to data transformation and optimization, particularly focusing on its support for echosounder data?

TahiriNadia replied:
What are the key stages in echodataflow's echosounder data processing pipeline, and how does each stage contribute to producing biological estimates such as abundance and biomass?

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+:::
+
+### Why Prefect?
+We chose Prefect among other Python workflow orchestration frameworks such as Apache Airflow[@airflow], Dagster[@dagster], Argo[@argo], Luigi[@luigi]. While most of these tools provide flexibily and level of abstraction to execute fisheries acoustics pipelines, we selected Prefect for the following reasons:
```
Add a space before parentheses. Airflow[@airflow], Dagster[@dagster], Argo[@argo], Luigi[@luigi]

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+The main goal of echodataflow is to allow users to configure an echosounder data processing pipeline through editing configuration “recipe” templates. Echodataflow can be configured through three templates: `datastore.yaml` which handles the data storage decisions, `pipeline.yml` which specifies the processing stages, and `logging.yaml` which sets the logging format. 
+
+### Data Storage Configuration
+In [Figure %s ](#fig:datastore_config): `datstore.yaml` we provide an example of a data store configuration for a ship survey. In this scenario we want to process data from the Joint U.S.-Canada Integrated Ecosystem and Pacific Hake Acoustic Trawl Survey [@NWFSC_FRAM_2022] which is being publicly shared on an AWS S3 bucket by NOAA National Centers for Environmental Information Acoustics Archive ( ...
```
Add coma before we in In [Figure %s ](#fig:datastore_config): `datstore.yaml` we

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+
+### Pipeline Configuration
+The pipeline configuration file’s purpose is to list the stages of the processing pipeline and the computational set-up for their execution. In [Figure %s ](#fig:pipeline_config) we provide an example with several stages: `open_raw`, `combine_echodata`, `compute_Sv`, `compute_MVBS`. Each stage is executed as a separate Prefect subflow (a component of a Prefect workflow), and one can specify addit ...
```
Add coma before we in In [Figure %s ](#fig:pipeline_config) we

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+
+## Workflow Execution
+To convert a scientific pipeline into an executable Prefect workflow, one needs to organize its components into flows, sublfows, and tasks (the key objects of Prefect’s execution logic). Usually, the stages of a pipeline will be organized into flows and subflows, while the individual pieces of work within the stage will be organized into tasks. In practice, flows, subflows, and tasks are all Pyth ...
```
Add coma before we in In [Figure %s ](#fig:flow_task) we

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+
+
+In [Figure %s ](#fig:log_output) we show an example of output logs for different processing stages.
```
Add coma before we in In [Figure %s ](#fig:log_output) we

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+* Basic Logging with Dask Worker Streams: this approach configures `dask` worker streams to handle `echodataflow` logs, which is straightforward if exact log order is not crucial.
+* Centralized Logging with Amazon CloudWatch [@cloudwatch]: this approach centralizes all logs for easy access and analysis. It can be useful when users are already utilizing AWS.
+* Advanced Logging with Apache Kafka [@kafka] and Elastic Stack [@elastic_stack](Elasticsearch, Kibana, Beats, Logstash): this approach leverages Kafka for log aggregation and Elastic Stack for log analysis and visualization, offering a robust solution for those who can maintain the infrastructure, for example data center managers.
+
```
Could you explain the advantages of using Amazon CloudWatch for centralized logging in echodataflow, especially for users already integrated into AWS environments?

TahiriNadia replied:
In the context of processing large data archives with echodataflow, how does a robust logging system contribute to identifying and resolving issues during data processing, particularly concerning compatibility with the pipeline?

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+
+## Example Use Case: Processing Ship Survey Data from an Archive
+
+We demonstrate a workflow of processing all acoustic data for the 2017 Joint U.S.-Canada Integrated Ecosystem and Pacific Hake Acoustic Trawl Survey through a few routine processing stages. The survey spans a period of 06/15/2017 - 09/13/2017, covering the entire west coast of the US and Canada. Figure 1(a) shows a typical map of the transect schedule of the survey. Raw acoustic data are collecte ...
```
Missing link to the Figure in Figure 1(a) shows a typical map of the transect schedule of the survey

TahiriNadia commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,268 @@
+We thank the Fisheries Engineering and Acoustic Technologies team at the NOAA Northwest Fisheries Science Center: Julia Clemons, Alicia Billings, Rebecca Thomas, Elizabeth Phillips for introducing us to the Pacific Hake Survey operations and biomass estimation workflow.
+
+## Funding:
+NOAA Fisheries, eScience Institute
```
The outstanding work and the clear, insightful text provided by the authors are greatly appreciated. Their contributions to advancing fisheries acoustics through echodataflow are commendable. The quality of the research and its implications for broader scientific communities are truly commendable.

valentina-s commented:
> Dear @valentina-s, I will list my comments one by one below in detail.
> 
> The submitted article explores the application of a recipe-based approach to dataflow orchestration within the fisheries context. This methodology aims to optimize data processing workflows in fisheries research and management by leveraging advanced scientific computing techniques. The article is highly relevant, well-structured, and written with clarity.

Thanks @TahiriNadia! I am looking forward to your comments, and will fill in missing DOI's per @ameyxd's comment.

mepa commented:
Hi @gol-danRuman and @TahiriNadia, the Proceedings Committee has extended the initial complete review deadline to Monday, July 8th. Thanks so much for volunteering to review and we look forward to seeing your comments!

TahiriNadia commented:
Dear @valentina-s et al.,

Thank you for the updates. Please tag me back in when the paper is ready for a re-review!

Cheers,

TahiriNadia commented on papers/valentina_staneva/case_study.png:
It would be great to have this table not as an image.

TahiriNadia commented on papers/valentina_staneva/datastore_config.png:
Please, change the next image as a snippets (listings).

TahiriNadia commented on papers/valentina_staneva/flow_task.png:
Please, change the next image as a snippets (listings).

TahiriNadia commented on papers/valentina_staneva/logging_config.png:
Please, change the next image as a snippets (listings).

TahiriNadia commented on papers/valentina_staneva/log_output.png:
Please, change the next image as a snippets (listings).

valentina-s committed:
fixing echoview references

valentina-s committed:
adding spaces before citations and fixing a few names

mepa commented:
Hi @valentina-s,  have you have a chance to look at the reviewer's comments on your paper? Just a heads up that it may take more than one iteration of review before the forthcoming [final author revision deadline](https://github.com/scipy-conference/scipy_proceedings/blob/2024/README.md#author-deadlines).

valentina-s committed:
converting images to code snippets

valentina-s committed:
text tuning

valentina-s committed:
adding extra explanation of dimensions and needs

valentina-s committed:
adding Jetstream references

valentina-s committed:
adding roles

valentina-s committed:
Adding acknowledgements

mepa commented:
Hey @valentina-s, I will be stepping in to provide a second review for this paper. I look forward to reviewing it and will plan to post comments later this evening.

valentina-s committed:
adding year to saildrone map

valentina-s committed:
adding comma

valentina-s committed:
polishing

valentina-s committed:
main update

valentina-s committed:
abstract

valentina-s committed:
abstract

valentina-s committed:
indenting abstract

TahiriNadia commented:
Dear @valentina-s,

Thank you for the updates. Please tag me back in when the paper is ready for a re-review!

Cheers,
Nadia

TahiriNadia commented:
Dear @valentina-s,

The error from Dask reference.
'DOI Exists: Citation does not have DOI: dask'

Best

TahiriNadia commented on papers/valentina_staneva/myst.yml:
```diff
@@ -0,0 +1,103 @@
+  error_rules:
+    - rule: doi-exists
+      severity: ignore
+      keys:
```
you can add dask reference on this list.

DOI Exists: Citation does not have DOI: dask

valentina-s committed:
adding dask to doi exception list

valentina-s committed:
removing . from formats

valentina-s committed:
correcting references when several together, ro long names

valentina-s committed:
fixing abstract issues

valentina-s committed:
fixing one more no doi reference

valentina-s commented:
@TahiriNadia Thanks for the review! I have converted the images to text snippets, corrected references, and added more explanation of the varying dimensions across deployments and the associated challenges in the end of the Motivation section. Now the checks are passing, but if some references do not look as expected let me know (should I leave the ones with n. d. as is?).

mepa commented on papers/valentina_staneva/case_study.png:
This .png is not in the paper. Did you intend to include this?

valentina-s replied:
I integrated this in the text.

mepa commented on papers/valentina_staneva/echodataflow_flow_runs.png:
Not in paper, did you intend to include this?

valentina-s replied:
updated with another figure

mepa commented on papers/valentina_staneva/figure1.png:
This is from the example paper and could be removed.

mepa commented on papers/valentina_staneva/figure2.png:
This is from the example paper and could be removed.

mepa commented on papers/valentina_staneva/task_progress.png:
Not in paper, did you intend to include this?

valentina-s replied:
I have another figure addressing this.

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+
+## Motivation
+Acoustic fisheries surveys and ocean observing systems collect terabytes of echosounder (water column sonar) data that require custom processing pipelines to obtain the distributions and abundance of fish and zooplankton in the ocean [@ncei_story_map]. The data are collected by sending an acoustic signal into the ocean which scatters from objects in the water column and the returning “echo” is re ...
```
"however, with limited computing resources" -> What is being contrasted here? Would "and with limited computing resources" fit better?

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+:::{figure} workflow_variations.png
+:label: fig:workflow_variations
+** Fisheries Acoustics Workflow Variations:** Various use cases (fisheries, data management, machine learning, education) drive different needs for data storage and computing infrastructure. Options are abundant but adopting new technology and adapting workflows across use cases is not trivial.
```
typo (extra space) in ** Fisheries Acoustics Workflow Variations:** -> **Fisheries Acoustics Workflow Variations:**

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+A **data manager** wants to process several terabytes of mooring data and serve them to the scientific community. They have a few Python scripts to do this for a small set of files at a time, but want to scale the processing for many deployments using a cloud infrastructure.
+
+A **machine learning engineer** is developing an ML algorithm to automatically detect fish on a USV. They need to prepare a large dataset for training but do not know all the necessary preprocessing steps. They are very familiar with ML libraries but do not have the domain-specific knowledge for acoustic data processing. They are also not familiar with distributed computing libraries.
```
Would moving the "machine learning engineer" paragraph above the "data manager" one correspond with the order of "Use Cases" icons in Figure 2? In other words, does the database icon refer to "data manager" and the brain/neuron-type icon to "machine learning engineer"?

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+A **machine learning engineer** is developing an ML algorithm to automatically detect fish on a USV. They need to prepare a large dataset for training but do not know all the necessary preprocessing steps. They are very familiar with ML libraries but do not have the domain-specific knowledge for acoustic data processing. They are also not familiar with distributed computing libraries.
+
+An **acoustics graduate student** obtained echosounder data analysis scripts from a retired scientist but do not have all the parameters needed to reproduce the results in order to proceed with their dissertation research.
```
```suggestion
An **acoustics graduate student** obtained echosounder data analysis scripts from a retired scientist but does not have all the parameters needed to reproduce the results in order to proceed with their dissertation research.
```

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+
+### Software Landscape
+Traditionally echosounder data processing pipelines are executed within a GUI-based software (e.g. Echoview [@echoview_software_pty_ltd_echoview_nodate], LSSS [@korneliussen_lsss_2006], ESP3 [@ladroit_esp3_2020], Matecho [@perrot_matecho_2018]. These software packages have been invaluable for onboard real-time visualization, as well as post-survey data screening and annotation. Some of them also  ...
```
missing a closing parenthesis to "(e.g. Echoview..."

mepa replied:
"with the goal to substitute...and make" -> "with the goal of substituting...and making..." - I'm not certain about correct grammar but the latter sounds more natural to me.

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+##### Data Storage
+
+Researchers are faced with decisions of where to store the data from experiments, intermediate products, and final results. Initially, data are usually stored on local hard drive storage associated with the instrument (which on some platforms may have limited capacity), but eventually, these data may be transferred to a data archive if one is maintained within the community. Some agencies (e.g. N ...
```
missing a closing parenthesis to "(e.g. NOAA..."

mepa replied:
"may require to store" -> "may require storing"

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+
+##### Computing Infrastructure
+
+With the growth of the echosounder datasets, researchers face challenges processing the data on their personal machines: both in terms of memory usage and computational time. A typical first attempt for resolution would be to amend the workflow to process smaller chunks of the data and parallelize operations across multiple cores if available. However, today researchers are also presented with a  ...
```
Should "distributing libraries" be "distributed libraries"?

mepa commented on papers/valentina_staneva/main.md:
```diff
@@ -0,0 +1,422 @@
+With the growth of the echosounder datasets, researchers face challenges processing the data on their personal machines: both in terms of memory usage and computational time. A typical first attempt for resolution would be to amend the workflow to process smaller chunks of the data and parallelize operations across multiple cores if available. However, today researchers are also presented with a  ...
+
+## Echodataflow Overview
+At the center of `echodataflow`'s design is the notion that a workflow can be configured through a set of recipes (`.yaml` files) that specify the pipeline, data storage, and logging structure. The idea draws inspiration from the Pangeo-Forge Project which facilitates the Extraction, Transformation, Loading (ETL) of earth science geospatial datasets from traditional repositories to analysis-ready ...
```
This is not necessary, but just a heads up that one can add abbreviations to the `myst.yml` file so that a reader can mouse over the abbreviation in the html version of the paper to see the full text.

For example,
```markdown
  abbreviations:
    ETL: Extraction, Transformation, Loading
    ARCO: analysis-ready, cloud-optimized
```

mepa reviewed (commented):
Hi @valentina-s, thanks for your patience as the proceedings committee looked for a second reviewer for your paper. I am the assigned editor for this paper and do not have a specific background in fisheries acoustics, but do have experience with workflow orchestration and using some of the same libraries for other tasks. 

My initial comments are somewhat fine-grained but I will go ahead and post the review so as not to delay any longer then follow up with broader feedback, if needed, in the coming days.

valentina-s committed:
removing template figures

valentina-s committed:
edits based on @mepa review

valentina-s committed:
some language polishing

valentina-s committed:
removing quotes in beginning of sentences

valentina-s committed:
removing unnecessary images

valentina-s committed:
reverting myst to see if checks pass

valentina-s committed:
adding back the abbreviations

valentina-s committed:
trying to clean invisible characters

valentina-s committed:
removing repeating ASW key

valentina-s committed:
adding echodataflow repo as a reference

valentina-s committed:
add echodataflow to no DOI keys

valentina-s commented:
Thanks @mepa for the review. I have addressed your comments.

mepa commented:
@valentina-s, thanks for making those updates.

Though I did not get the chance to run echodataflow and potentially provide further feedback before the author revision deadline, I read the paper over again and do feel it is ready for inclusion in the Proceedings.

mepa commented:
@TahiriNadia, thanks very much for reviewing! Do you approve this paper for inclusion in the Proceedings?
